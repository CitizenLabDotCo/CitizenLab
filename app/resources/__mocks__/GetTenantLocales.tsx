import { GetAppConfigurationLocalesChildProps } from 'resources/GetAppConfigurationLocales';

export const mockGetTenantLocales = ['en', 'fr-BE'] as GetAppConfigurationLocalesChildProps;
