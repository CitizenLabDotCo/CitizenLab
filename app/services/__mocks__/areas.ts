import { BehaviorSubject } from 'rxjs';
import { IIdea } from 'services/ideas';

let mockIdea: IIdea | null = null;

export const __setMockIdea = (idea: IIdea) => {
  mockIdea = idea;
};

export const areaByIdStream = jest.fn((_areaId) => {
  const observable = new BehaviorSubject(mockIdea);
  return {
    observable
  };
});
