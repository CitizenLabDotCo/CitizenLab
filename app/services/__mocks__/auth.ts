import { IUser } from 'services/users';
export const mockUser = {
  data: {
    id: '522ae8cc-a5ed-4d31-9aa0-470904934ec6',
    type: 'users',
    attributes: {
      first_name: 'Test',
      last_name: 'Citizenlab',
      slug: 'test-citizenlab',
      locale: 'en',
      avatar: {
        small: null,
        medium: null,
        large: null
      },
      roles: [],
      bio_multiloc: {},
      registration_completed_at: '2018-11-26T15:40:54.355Z',
      invite_status: null,
      created_at: '2018-11-26T15:41:19.782Z',
      updated_at: '2018-11-26T15:41:19.782Z',
      email: 'test@citizenlab.co',
      custom_field_values: {
        birthyear: 1990,
        domicile: 'outside',
        gender: 'female',
        politician: 'retired_politician'
      },
      unread_notifications: 0
    },
    relationships: {
      granted_permissions: {
        data: []
      }
    }
  }
} as IUser;

const adminRole = [{ type: 'admin' }];

function setRoles(roles) {
  const user = mockUser;
  user.data.attributes.roles = roles;
  return user;
}
export const mockAdmin = setRoles(adminRole) as IUser;

export const mockProjectModerator = (projectId: string) => setRoles([{ type: 'project_moderator', project_id: projectId }]) as IUser;
