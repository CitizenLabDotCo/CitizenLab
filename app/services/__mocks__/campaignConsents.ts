export const updateConsentByCampaignIDWIthToken = jest.fn().mockImplementation(() => Promise.resolve());
