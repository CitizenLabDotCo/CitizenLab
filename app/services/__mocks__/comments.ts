import { ICommentData } from 'services/comments';
import { BehaviorSubject } from 'rxjs';

export const makeCommentData = (attributes = {}, relationships = {}) : ICommentData => ({
  id: 'commentId',
  type: 'comments',
  attributes: {
    body_multiloc: { en: 'Just a comment' },
    publication_status: 'published',
    upvotes_count: 0,
    downvotes_count: 0,
    created_at: '2019-03-12T00: 00: 00.000Z',
    updated_at: '2019-03-26T14: 32: 32.000Z',
    children_count: 0
  }
  relationships: {
    idea: {
      data: {
        id: 'ideaId',
        type: 'ideas'
      }
    },
    author: {
      data: {
        id: 'authorId',
        type: 'users'
      }
    },
    user_vote: {
      data: null
    }
  }
});
