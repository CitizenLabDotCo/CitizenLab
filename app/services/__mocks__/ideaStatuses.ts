export function makeStatus(id) {
  return {
    id,
    type: 'idea_status',
    attributes: {
      title_multiloc: { en: 'Just a status' },
      color: 'blue',
      code: id,
      ordering: 0,
      description_multiloc: { en: 'The status of an idea that has a status' }
    }
  };
}
