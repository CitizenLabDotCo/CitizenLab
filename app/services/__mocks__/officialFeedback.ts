import { IOfficialFeedback } from 'services/officialFeedback';

export const mockOfficialFeedback = {
  data: [
    {
      type: 'official_feedbacks',
      id: 'feedbackId1',
      attributes: {
        author_multiloc: {
          en: 'Testing Official Department',
        },
        body_multiloc: {
          en: 'Update: Do Not Keep Calm, please panic now. End of transmission.',
        },
        created_at: 'Some time ago',
        updated_at: 'Some shorter time ago'
      }
    },
    {
      type: 'official_feedbacks',
      id: 'feedbackId2',
      attributes: {
        author_multiloc: {
          en: 'Testing Official Department',
        },
        body_multiloc: {
          en: 'This is an important official communication from the testing department. Keep calm and read on.',
        },
        created_at: 'Some longer time ago',
        updated_at: 'Some longer shorter time ago'
      }
    }
  ]
} as IOfficialFeedback;
