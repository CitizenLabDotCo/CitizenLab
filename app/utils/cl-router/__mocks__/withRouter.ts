import { WithRouterProps } from 'react-router';

export const mockWithRouterProps : WithRouterProps = {
  location: {
    pathname: 'some-page',
    search: '',
    query: '',
    state: '',
    action: 'PUSH',
    key: '',
    hash: ''
  },
  router: {
    push: () => null,
    replace: () => null,
    go: () => null,
    goBack: () => null,
    goForward: () => null,
    setRouteLeaveHook: () => null,
    createPath: () => '',
    createHref: () => '',
    isActive: () => true,
  },
  params: {},
  routes: []
};
