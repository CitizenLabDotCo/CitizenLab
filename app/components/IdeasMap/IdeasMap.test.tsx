import React from 'react';
import { shallow } from 'enzyme';
import 'jest-styled-components';
import { mockWithRouterProps } from '../../utils/cl-router/__mocks__/withRouter';
import leaflet from 'leaflet';
import { IdeasMap } from '.';

jest.mock('leaflet');

describe('IdeasMap', () => {
  it('renders correctly', () => {
    const wrapper = shallow(
      <IdeasMap
        ideaMarkers={[]}
        projectIds={['379cac45-d26b-505c-9215-b2677a693d37']}
        phaseId="c43968d3-171e-570c-8b71-f5fc0e55bbfa"
        {...mockWithRouterProps}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('calls leaflet.popup() when child Map calls onMapClick', () => {
    const wrapper = shallow(
      <IdeasMap
        ideaMarkers={[]}
        projectIds={['379cac45-d26b-505c-9215-b2677a693d37']}
        phaseId="c43968d3-171e-570c-8b71-f5fc0e55bbfa"
        {...mockWithRouterProps}
      />
    );
    wrapper.find('IdeasMap__StyledMap').simulate('click');
    expect(leaflet.popup).toBeCalledTimes(1);
  });
});
