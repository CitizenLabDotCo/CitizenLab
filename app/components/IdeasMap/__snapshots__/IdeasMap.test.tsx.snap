// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IdeasMap renders correctly 1`] = `
<IdeasMap__Container>
  <accessibility__ScreenReaderOnly>
    <FormattedMessage
      defaultMessage="Map with overview of ideas based on location"
      id="app.components.IdeasMap.mapTitle"
    />
  </accessibility__ScreenReaderOnly>
  <IdeasMap__StyledMap
    boxContent={null}
    fitBounds={true}
    onBoxClose={[Function]}
    onMapClick={[Function]}
    onMarkerClick={[Function]}
    points={Array []}
  />
  <div
    className="create-idea-wrapper"
  >
    <Component
      onClick={[Function]}
      participationContextType="phase"
      phaseId="c43968d3-171e-570c-8b71-f5fc0e55bbfa"
      projectId="379cac45-d26b-505c-9215-b2677a693d37"
    />
  </div>
</IdeasMap__Container>
`;
